pipeline/schemas/pipeline.py

```diff
@@ -88,3 +88,15 @@ class PipelineCreate(BaseModel):
     # By default a Pipeline will require GPU resources
     compute_type: ComputeType = ComputeType.gpu
     compute_requirements: Optional[ComputeRequirements]
+
+    @validator("compute_requirements")
+    def compute_type_is_gpu(cls, v, values):
+        """If compute_type is not 'gpu' we don't expect any additional compute
+        requirements to be specified.
+        """
+        if values["compute_type"] != ComputeType.gpu:
+            if v and v.min_gpu_vram_mb:
+                raise ValueError(
+                    "min_gpu_vram_mb should only be specified for gpu workloads"
+                )
+        return v
```

Comment:
Neat way of codifying the constraints. Hopefully we can design the OpenAPI spec in a way that the generated schemas contain such checks.